### PR TITLE
use Redis as message broker and backend store for EditorChannelServer; resolves #1101

### LINKED ISSUE #1101: Set up message broker for transient messages
Currently, transient messages are relayed by the Node process, they need to be relayed by a third party in order to allow scaling a room across multiple backend nodes. 

It's just a simple pub-sub broker. Use RabbitMQ? Build in-house?

### PR DESCRIPTION
**Pull request by kschzt**
- add pubsub channels via Redis
- store edit logs and chat in Redis instead of RethinkDB
- migration script to move edit logs & chat to Redis
- prepare to stop using RethinkDB altogether in the next release

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ Be sure to [watch the tutorials](http://bit.do/vizor) and [read the documentatio
 
 ### Installing
 
-Installing a local instance of Vizor requires [MongoDB](http://mongodb.org), [RethinkDB](http://rethinkdb.com) and either [node.js](https://nodejs.org) or [io.js](https://iojs.org/). To install the required packages, issue the following commands (on Mac using Homebrew):
+Installing a local instance of Vizor requires [MongoDB](http://mongodb.org), [Redis](http://redis.ui) and either [node.js](https://nodejs.org) or [io.js](https://iojs.org/). To install the required packages, issue the following commands (on Mac using Homebrew):
```
Comment (gmarinov) [inline README.md]: redis.io

File: lib/editorChannelServer.js [hunk 2]
```diff
@@ -500,21 +529,19 @@ EditorChannelServer.prototype.leaveChannel = function(channelName, client) {
 EditorChannelServer.prototype.sendToChannel = function(channelName, message, exceptId) {
```
Comment (gmarinov) [inline lib/editorChannelServer.js]: does it still need `, exceptId` ?

Comment (kschzt) [inline lib/editorChannelServer.js]: @gmarinov nope, removing

File: Dockerfile [hunk 3]
```diff
@@ -4,6 +4,7 @@ ENV HOME /root
 ENV NODE_ENV production
 ENV ENGI_BIND_IP 0.0.0.0
 ENV RETHINKDB_HOST rethink
```
Comment (kkananen) [inline Dockerfile]: is this line still needed?

**Comment by kkananen** [comment]
lgtm

File: Dockerfile (hunk 3 above)
Comment (kschzt) [inline Dockerfile]: @kkananen yes, for the migration from Rethink to be done

**Comment by kkananen** [comment]
lgtm

**Pull request merged**